src/player.js

```diff
@@ -144,6 +144,7 @@ class Player extends FakeEventTarget {
       let sources = config.sources;
       for (let i = 0; i < sources.length; i++) {
         if (Html5.canPlayType(sources[i].mimetype)) {
+          this.dispatchEvent(new FakeEvent(CustomEvents.SOURCE_SELECTED, {selectedSource: sources[i]}));
```

Comment:
Kaltura player should decorate the url with the session id. therefore, the source has to passed by reference. primitive like string are passed by value.